warehouse/forklift/legacy.py

```diff
@@ -1354,6 +1262,40 @@ def file_upload(request):
                 },
             )
 
+        # If the user provided attestations, verify and store them
+        if "attestations" in request.POST:
+            attestation_service = request.find_service(
+                IAttestationsService, context=None
+            )
+
+            try:
+                attestations: list[Attestation] = attestation_service.parse(
+                    attestation_data=request.POST["attestations"],
+                    distribution=Distribution(
+                        name=filename, digest=file_hashes["sha256"]
+                    ),
+                )
+            except AttestationUploadError as e:
+                raise _exc_with_message(
+                    HTTPBadRequest,
+                    str(e),
+                )
+
+            attestations_db_models = attestation_service.persist_attestations(
```

Comment:
```suggestion
            attestations_db_models = attestation_service.persist(
```

We might want to do a little more testing here to catch things like this!